metals/src/main/scala/scala/meta/internal/metals/testProvider/BuildTargetUpdate.scala

```diff
@@ -0,0 +1,39 @@
+package scala.meta.internal.metals.testProvider
+
+import org.eclipse.{lsp4j => l}
+
+final case class BuildTargetUpdate(
+    targetName: String,
+    targetUri: String,
+    events: java.util.List[TestExplorerEvent]
+)
+
+sealed abstract class TestExplorerEvent(val kind: String) {
+  def fullyQualifiedClassName: String
+  def className: String
+}
+object TestExplorerEvent {
+  final case class RemoveTestSuite(
+      fullyQualifiedClassName: String,
+      className: String
+  ) extends TestExplorerEvent("removeSuite")
+
+  final case class AddTestSuite(
+      fullyQualifiedClassName: String,
+      className: String,
+      symbol: String,
+      location: l.Location
+  ) extends TestExplorerEvent("addSuite")
+
+  final case class AddTestCases(
```

Comment:
I haven't looked too closely at the VS Code implementation yet, but do we need to have more info here about the parent? I know we have the `fullyQualifiedName`, but if we have for example

```
targetA
  suiteA
  suiteB
target B
  suiteB
```
And then we get a `AddTestCases` event. We have the `targetName` that it belongs to (we'll say targetA), but not the `suite`. In this scenario wouldn't we also need to know the `suite` that it belongs to in order to correctly place it without having to parse the `fullyQualifiedName`?

Reply:
Currently, `vscode.TestItem` (which is a basic block used in testing) has an id which consist of `build_target_id#fullyQualifiedName`. `TestItem`s form a tree data structure:
```
app
└── app#a
    ├── app#a.b.Suite1
    └── app#a.b
        ├── app#a.b.Suite2
        └── app#a.b.Suite3
```
So in my case, I have to parse `fullyQualifiedName` and at the same time it's sufficient to correctly place test cases. The only thing which can help will be to preprocess `fullyQualifiedName` into list of prefixes `[a, a.b, a.b.Suite1]` but I don't know if it will be useful for other clients.